cassovary-core/src/main/scala/com/twitter/cassovary/graph/node/ArrayBasedDirectedNode.scala

```diff
@@ -23,15 +23,16 @@ object ArrayBasedDirectedNode {
    * @param nodeId an id of the node
    * @param neighbors a seq of ids of the neighbors read from file
    * @param dir the stored graph direction (OnlyIn, OnlyOut, BothInOut or Mutual)
+   * @param sortedNeighbors true if the neighbors of the node will be sorted
    *
    * @return a node
    */
-  def apply(nodeId: Int, neighbors: Array[Int], dir: StoredGraphDir) = {
+  def apply(nodeId: Int, neighbors: Seq[Int], dir: StoredGraphDir, sortedNeighbors: Boolean = false) = {
     dir match {
       case StoredGraphDir.OnlyIn | StoredGraphDir.OnlyOut | StoredGraphDir.Mutual =>
-        UniDirectionalNode(nodeId, neighbors, dir)
+        UniDirectionalNode(nodeId, neighbors, dir, sortedNeighbors)
       case StoredGraphDir.BothInOut =>
-        BiDirectionalNode(nodeId, neighbors)
+        BiDirectionalNode(nodeId, neighbors, sortedNeighbors)
```

Comment:
may be just let a be of type FillingInEdgesBiDirectionalNode ?


On Wed, Feb 4, 2015 at 7:33 AM, Szymon <notifications@github.com> wrote:

> In
> cassovary-core/src/main/scala/com/twitter/cassovary/graph/node/ArrayBasedDirectedNode.scala
> <https://github.com/twitter/cassovary/pull/133#discussion_r24091099>:
>
> >        case StoredGraphDir.BothInOut =>
> > -        BiDirectionalNode(nodeId, neighbors)
> > +        BiDirectionalNode(nodeId, neighbors, sortedNeighbors)
>
> We would still need casting because cannot do:
>
> val a: Array[BiDirectionalNode] = new Array[FillingInEdgesBiDirectionalNode]
>
> Ok, will change it.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/twitter/cassovary/pull/133/files#r24091099>.
>